### ISSUE TITLE
ERROR:Unexpected segmentation fault encountered in worker.

### ISSUE THREAD

**Issue by JL-52Hertz** [ultralytics/ultralytics#22222]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

<img width="1281" height="481" alt="Image" src="https://github.com/user-attachments/assets/d0be1a1f-3cf1-42c0-a0c1-67445c68eac8" />
I'm running the same code on Server A and Server B, but Server B occasionally encounters the aforementioned error. Server A never experiences this error. I've checked Server B's hardware and found no issues. Is this a Yolo code optimization issue?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JL-52Hertz, thanks for opening this issue and for your interest in Ultralytics 🚀! This is an automated response to help you get support faster. An Ultralytics engineer will review and assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

Since you’re seeing a segmentation fault on one server but not the other, please include the following to speed up triage:
- Exact command(s) or code snippet used
- Full console output and any stack trace/logs
- OS, Python, PyTorch, CUDA, and CUDNN versions
- GPU model(s) and driver version
- ultralytics version (`pip show ultralytics`)
- Dataset/sample inputs if relevant
- Any differing settings between Server A and B (num_workers, multiprocessing start method, batch size, pinned memory, environment variables, etc.)
- Whether the issue reproduces with a small MRE and a single DataLoader worker (`num_workers=0`)

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?

**Comment by JL-52Hertz** [comment]
> Can you provide the output after running this command in terminal: `yolo checks`?

<img width="1020" height="625" alt="Image" src="https://github.com/user-attachments/assets/63da932a-1988-4319-a96b-4f1eed9cbd66" />

**Comment by Y-T-G** [comment]
You can try upgrading PyTorch

**Comment by JL-52Hertz** [comment]
> You can try upgrading PyTorch

Thank you,I will try it.

**Comment by glenn-jocher** [comment]
Sounds good—if the segfault still appears after upgrading PyTorch/Torchvision, please help us isolate it and share text logs (not screenshots) with these quick checks:

- Force single-process data loading to rule out DataLoader multiprocessing:
```bash
yolo task=detect mode=val model=yolo11n.pt data=coco8.yaml batch=4 workers=0
```
- Capture a full stack trace:
```bash
TORCH_SHOW_CPP_STACKTRACES=1 yolo task=detect mode=val model=yolo11n.pt data=coco8.yaml workers=0
```
- From Python, include detailed diagnostics from your currently imported package:
```python
from ultralytics.utils.checks import collect_system_info, check_torchvision, check_multiple_install
collect_system_info(); check_torchvision(); check_multiple_install()
```

Please paste all resulting text outputs and the complete traceback. If you’re running in Jupyter, restart the kernel after the upgrade. For general guidance see the Troubleshooting Common YOLO Issues doc and, if needed, how to provide a Minimal Reproducible Example so we can reproduce on our side: [Troubleshooting Common YOLO Issues](https://docs.ultralytics.com/guides/yolo-common-issues/) • [Creating a Minimal Reproducible Example](https://docs.ultralytics.com/help/minimum-reproducible-example/).

**Comment by JL-52Hertz** [comment]
> Sounds good—if the segfault still appears after upgrading PyTorch/Torchvision, please help us isolate it and share text logs (not screenshots) with these quick checks:
> 
> * Force single-process data loading to rule out DataLoader multiprocessing:
> 
> yolo task=detect mode=val model=yolo11n.pt data=coco8.yaml batch=4 workers=0
> * Capture a full stack trace:
> 
> TORCH_SHOW_CPP_STACKTRACES=1 yolo task=detect mode=val model=yolo11n.pt data=coco8.yaml workers=0
> * From Python, include detailed diagnostics from your currently imported package:
> 
> from ultralytics.utils.checks import collect_system_info, check_torchvision, check_multiple_install
> collect_system_info(); check_torchvision(); check_multiple_install()
> Please paste all resulting text outputs and the complete traceback. If you’re running in Jupyter, restart the kernel after the upgrade. For general guidance see the Troubleshooting Common YOLO Issues doc and, if needed, how to provide a Minimal Reproducible Example so we can reproduce on our side: [Troubleshooting Common YOLO Issues](https://docs.ultralytics.com/guides/yolo-common-issues/) • [Creating a Minimal Reproducible Example](https://docs.ultralytics.com/help/minimum-reproducible-example/).

Thank you.I have made the corresponding changes, because this error is accidental, even when using the previous environment. There is no error yet. Once the same error is reported, I will report it as soon as possible!

**Comment by glenn-jocher** [comment]
Thanks for the update—since it’s intermittent, please run your next session with these env vars so the crash yields a synchronous, readable trace, then paste the full text logs and the text output of `yolo checks` from Server B:

```bash
CUDA_LAUNCH_BLOCKING=1 PYTHONFAULTHANDLER=1 <your yolo/python command>
```
